module/VuFind/src/VuFind/Marc/Serialization/MarcXml.php

```diff
@@ -48,29 +69,72 @@ class MarcXml implements SerializationInterface
     public static function canParse(string $marc): bool
     {
         // A pretty naïve check, but it's enough to tell the different formats apart
-        return strncmp($marc, '<', 1) === 0;
+        return strncmp(trim($marc), '<', 1) === 0;
+    }
+
+    /**
+     * Check if the serialization class can parse the given MARC collection string
+     *
+     * @param string $marc MARC
+     *
+     * @return bool
+     */
+    public static function canParseCollection(string $marc): bool
+    {
+        // A pretty naïve check, but it's enough to tell the different formats apart
+        return strncmp(trim($marc), '<', 1) === 0;
+    }
+
+    /**
+     * Check if the serialization class can parse the given MARC collection file
+     *
+     * @param string $file File name
+     *
+     * @return bool
+     */
+    public static function canParseCollectionFile(string $file): bool
+    {
+        if (false === ($f = @fopen($file, 'r'))) {
+            throw new \Exception("Cannot open file '$file' for reading");
+        }
+        do {
+            $s = ltrim(fgets($f, 10));
+        } while (!$s);
```

Comment:
What happens if you feed this a file containing only whitespace? Do we need extra logic to break out of the loop at EOF?

Reply:
Oh, bad things probably. I'll need to fix that.

Reply:
Probably worth another test fixture! :-) In any case, I think once you get through that, I can get down to the remaining testing work.

Reply:
This is now fixed, so no endless loop. However, I'm undecided on whether an empty file should be considered unknown format or ISO2709. Currently it is considered unknown, but it could be argued that an empty file is a valid one, just without any records.

Reply:
I have no problem with leaving it "Unknown" for now -- I don't think there are many scenarios where somebody would intentionally pass in an empty file, so treating that as an error rather than a valid but empty collection seems more appropriate. I suppose it might be annoying to get intermittent errors if there's an automated process that sometimes generates records and sometimes generates empty files... but if a process operating on an empty file fails and does nothing, the net result as the same as if it succeeds and does nothing, so I don't think it really matters.